libs/metrics/cdc_transmission_levels.py

```diff
@@ -0,0 +1,115 @@
+        level_high = math.inf
+
+    if value is None or math.isinf(value) or np.isnan(value):
+        return TransmissionLevel.UNKNOWN
+
+    if value < level_low:
+        return TransmissionLevel.LOW
+    elif value < level_moderate:
+        return TransmissionLevel.MODERATE
+    elif value < level_substantial:
+        return TransmissionLevel.SUBSTANTIAL
+
+    return TransmissionLevel.HIGH
+
+
+def case_density_transmission_level(value: float) -> TransmissionLevel:
+    # CDC thresholds are number of cases over a 7 day average. Our case density
+    # metrics are cases over a 7 day average.  To square the two, we should divide
+    # CDC thresholds by 7 to get the equivalent threshold for our 7-day averages.
+    thresholds = [10 / 7.0, 50 / 7.0, 100 / 7.0]
+    return calc_transmission_level(value, thresholds)
+
+
+def test_positivity_transmission_level(value: float) -> TransmissionLevel:
+    thresholds = [0.05, 0.08, 0.10]
+    return calc_transmission_level(value, thresholds)
+
+
+def top_level_transmission_level(
+    case_density_level: TransmissionLevel, test_positivity_level: TransmissionLevel,
+) -> TransmissionLevel:
+    """Calculate the overall risk for a region based on metric risk levels."""
+
+    level_list = [
+        test_positivity_level,
+        case_density_level,
+    ]
+
+    if TransmissionLevel.HIGH in level_list:
+        return TransmissionLevel.HIGH
+    elif TransmissionLevel.SUBSTANTIAL in level_list:
+        return TransmissionLevel.SUBSTANTIAL
+    elif TransmissionLevel.MODERATE in level_list:
+        return TransmissionLevel.MODERATE
+    elif TransmissionLevel.UNKNOWN in level_list:
+        return TransmissionLevel.UNKNOWN
+
+    return TransmissionLevel.LOW
+
+
+def calculate_transmission_level_from_metrics(
+    metrics: can_api_v2_definition.Metrics,
+) -> can_api_v2_definition.CDCTransmissionLevel:
+    case_density_level = case_density_transmission_level(metrics.caseDensity)
+    test_positivity_level = test_positivity_transmission_level(metrics.testPositivityRatio)
+
+    overall_level = top_level_transmission_level(case_density_level, test_positivity_level)
+    return overall_level
+
+
+def calculate_transmission_level_from_row(row: pd.Series):
+
+    case_density_level = case_density_transmission_level(row[MetricsFields.CASE_DENSITY_RATIO])
+    test_positivity_level = test_positivity_transmission_level(row[MetricsFields.TEST_POSITIVITY])
+
+    return top_level_transmission_level(case_density_level, test_positivity_level)
+
+
+def calculate_transmission_level_timeseries(
+    metrics_df: pd.DataFrame, metric_max_lookback=top_level_metrics.MAX_METRIC_LOOKBACK_DAYS
+):
+    metrics_df = metrics_df.copy()
+    # Shift infection rate to align with infection rate delay
+    infection_rate = metrics_df[MetricsFields.INFECTION_RATE].shift(
+        top_level_metrics.RT_TRUNCATION_DAYS
+    )
+    metrics_df[MetricsFields.INFECTION_RATE] = infection_rate
```

Comment:
Uh, do we need this infection_rate stuff?

Reply:
No - I can actually delete these functions because we're not doing timeseries transmissions levels for now (would be easy enough to add in in the future if we want)